src/compartment.c

```diff
@@ -413,6 +450,12 @@ parse_lib_file(char *lib_name, struct Compartment *new_comp)
     new_lib->rela_maps_count = 0;
     new_lib->rela_maps = NULL;
 
+    new_lib->tls_sec_addr = 0x0;
+    new_lib->tls_sec_size = -1;
```

Comment:
These are unsigned `size_t` so what does `-1` mean in that case? I think this might be UB.

Reply:
I believe it's valid to underflow unsigned types. This will just mean `MAX_SIZE`. So I don't think it's UB, but it's shoddy logic, as there could perhaps be a TLS section of size `MAX_SIZE`.

Reply:
I do think that assigning negative values is probably UB, and given that C doesn't mandate two's complement, one couldn't guarantee it being `MAX_SIZE`.

Reply:
For the 2018 standard, 6.3.1.3.2 (page 38 in https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2310.pdf):
> Otherwise, if the new type is unsigned, the value is converted by repeatedly adding or subtracting
one more than the maximum value that can be represented in the new type until the value is in the
range of the new type.61)

If I understand this correctly, `(size_t) -1` should then be `-1 + 1 * (SIZE_MAX + 1)` (we only need to add one more than the maximum value once).

Anyway, I think a default value of `0` should also work here, so I could replace this with that, if you think that's better.

Reply:
> Anyway, I think a default value of 0 should also work here, so I could replace this with that, if you think that's better.

Yes please :)

Reply:
Changed in https://github.com/capablevms/CHERI-ELF-comp/pull/28/commits/3b177988faefa505ee0813120a10ab17d4914fa7.